lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -18,18 +18,27 @@ def initialize(options={})
       # @options = DEFAULT_OPTIONS.merge(options)
       @options = options
       @client = build_client(@options)
+      # mutex to prevent requests and sniffing to access the
+      # connection pool at the same time
+      @sniff_request_mutex = Mutex.new
       start_sniffing!
     end
 
     def template_install(name, template, force=false)
-      if template_exists?(name) && !force
-        @logger.debug("Found existing Elasticsearch template. Skipping template management", :name => name)
-        return
+      @sniff_request_mutex.synchronize do
+        if template_exists?(name) && !force
+          @logger.debug("Found existing Elasticsearch template. Skipping template management", :name => name)
+          return
+        end
+        template_put(name, template)
       end
-      template_put(name, template)
     end
 
     def bulk(actions)
+      @sniff_request_mutex.synchronize { synchronized_bulk(actions) }
+    end
+
+    def synchronized_bulk(actions)
```

Comment:
Errr, this is a weird method name because it is in fact not synchronized. Maybe `unsafe_bulk` or `unsynchronized_bulk` would be more clear?